src/v/cloud_storage/remote_segment.cc

```diff
@@ -219,6 +219,12 @@ remote_segment::offset_data_stream(
                   .file_pos = 0,
                 });
     }
+    vlog(
+      _ctxlog.debug,
+      "Offset data stream start reading at {}, log offset {}, delta {}",
```

Comment:
I was thinking about an alternative solution - can we for a single partition read save somewhere next rp offset and pass it to the segment reader. The segment reader will then just skip every batch before this offset. This way overlaps shouldn't be a problem.